### PR TITLE
Improvement phase 2

### PR DESCRIPTION
**Pull request by mfdeveloper508**
- support 005010X224A2
- updating rules of EDI 837P
- adding new segments

### CODE REVIEW

File: examples/example_5010_837d/example.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	if err = newChange.Validate(nil); err != nil {
+		util.Log().Error().Log(err.Error())
+		return
+	}
+
+	err = newChange.Validate(&InterchangeRule)
+	if err = newChange.Validate(nil); err != nil {
+		util.Log().Error().Log(err.Error())
```
Comment (adamdecaf) [inline examples/example_5010_837d/example.go]: Let's create a logger in each of these examples. Having libraries log is not good practice. We don't want the util.Log() function to be used on accident.

File: pkg/file/transactionset.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (r *TransactionSet) Validate(transRule *rules.TransactionRule) error {
 
 				if err = r.Loops[segIndex].Validate(&rule); err != nil {
 					if rules.IsMaskRequired(rule.Mask) {
-						log.Println(err)
+						util.Log().Debug().Log("Validation Error: " + err.Error())
 						return fmt.Errorf("loop(%02d) should have valid %s loop, %s", segIndex, strings.ToLower(rule.Name), err.Error())
```
Comment (adamdecaf) [inline pkg/file/transactionset.go]: The library should return errors wrapped with an appropriate description. Let's remove the logging from `pkg/...`

**Review by adamdecaf — COMMENTED** [review]
This code looks great. Thank you for putting in the word to implement this!

File: pkg/loops/composite_loop.go [hunk 3]
```diff
@@ -129,6 +131,10 @@ func (r *CompositeLoop) Parse(data string, args ...string) (int, error) {
 				r.SubLoops = append(r.SubLoops, *newChild)
 			} else {
 				if repeatIdx == 0 && rules.IsMaskRequired(rule.Mask) {
+					length := util.GetRecordSize(data[read:])
+					if length > 0 {
+						log.Println("Parse Error:(" + rule.Name + "), (" + err.Error() + "), (" + data[read:read+int(length)] + ")")
```
Comment (adamdecaf) [inline pkg/loops/composite_loop.go]: Library code shouldn't log. Can we return an error instead? 

```suggestion
						fmt.Errorf("%s parsing of %s failed: %v", rule.Name, data[read:read+int(length)], err)
```

Comment (mfdeveloper508) [reply pkg/loops/composite_loop.go]: sorry, make sense

Comment (mfdeveloper508) [reply pkg/loops/composite_loop.go]: fixed again

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**